### PR TITLE
added wine model

### PR DESCRIPTION
**Pull request by AmalieSonderskov**

### CODE REVIEW

**Review by Hindubedu — COMMENTED** [review]
Maybe we can consider if we wanna use DateTime?

**Comment by simonitminds** [comment]
image

**Comment by simonitminds** [comment]
https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExdXczY2kxb3FlYjJtNHl1YTduNTM4bWh5bnJjcG55M2l0b2p3bWRyeCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/l2SpUoAPo0CBOkyxq/giphy.gif

**Comment by simonitminds** [comment]
<div style="width:100%;height:0;padding-bottom:69%;position:relative;"><iframe src="https://giphy.com/embed/l2SpUoAPo0CBOkyxq" width="100%" height="100%" style="position:absolute" frameBorder="0" class="giphy-embed" allowFullScreen></iframe></div><p><a href="https://giphy.com/gifs/hulu-spaced-digital-rights-group-l2SpUoAPo0CBOkyxq">via GIPHY</a></p>

**Comment by simonitminds** [comment]
image

**Review by Hindubedu — COMMENTED** [review]
??

**Review by Hindubedu — COMMENTED** [review]
hrhgrgrhtr

**Review by Hindubedu — APPROVED** [review]

File: Domain/Wine.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace WineCellar.Domain;
+
+public class Wine
+{
+    public int Id { get; set; }
+
+    [Required]
```
Comment (simonitminds) [inline Domain/Wine.cs:9]: we use fluent validation. https://github.com/CarterCommunity/Carter

i examples viser de hvordan kan man bruge fluent validation til at validere modeller

File: Feature/Wines/WineModule.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        app.MapGet(
+                "/wines",
+                (HttpContext context, ApplicationDbContext dbContext) =>
+                {
+                    var user = context.User;
+                    if (user is null)
+                    {
+                        throw new UnauthorizedAccessException();
+                    }
+                    var name = context.User.Identity?.Name;
```
Comment (simonitminds) [inline Feature/Wines/WineModule.cs:24]: tjek det her sammen med ovenstaaende, giver ikke mening at tjekke p[ en nullable value 2 gange

File: Feature/Wines/WineModule.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                    var wines = dbContext.Users.FirstOrDefault(x => x.Username == name)?.Wines;
+
+                    return wines;
+                }
+            )
+            .Produces<List<Wine>>()
```
Comment (simonitminds) [inline Feature/Wines/WineModule.cs:30]: Burder der ikke v;re et RequireAuth her?

File: Persistence/EntityTypeConfigurations/UserConfiguration.cs [hunk 4]
```diff
@@ -11,5 +11,10 @@ public void Configure(EntityTypeBuilder<User> builder)
         builder.HasKey(e => e.Id);
         builder.HasIndex(e => e.Username).IsUnique();
         builder.Property(e => e.Username).HasMaxLength(50);
+        builder
```
Comment (simonitminds) [inline Persistence/EntityTypeConfigurations/UserConfiguration.cs:14]: rigtig fint

File: Persistence/EntityTypeConfigurations/WineConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+public class WineConfiguration : IEntityTypeConfiguration<Wine>
+{
+    public void Configure(EntityTypeBuilder<Wine> builder)
+    {
+        builder.HasKey(e => e.Id);
+        builder.HasIndex(e => e.Name);
+        builder.Property(e => e.Name).HasMaxLength(50);
+        builder.HasIndex(e => e.Year);
+        builder.HasIndex(e => e.Type);
```
Comment (simonitminds) [inline Persistence/EntityTypeConfigurations/WineConfiguration.cs:15]: vil gerne i explicit satte userid key ker

**Review by simonitminds — COMMENTED** [review]
lidt at forbedre

**Pull request merged**